### PR TITLE
hyperbahn/handler: retry the send relay ad endpoint

### PR DESCRIPTION
**Pull request by Raynos**
We noticed that we are getting socket errors during
restarts of our hyperbahn cluster.

This adds two retries with upto 4s of latency to the
advertisement logic

This allows us to smooth over socket errors that
occur because of restarts as part of deploys.

r: @jcorbin @kriskowal

### CODE REVIEW

**Comment by kriskowal** [comment]
:ok:

File: node/hyperbahn/handler.js [hunk 1]
```diff
@@ -28,6 +28,9 @@ var util = require('util');
 var TChannelJSON = require('../as/json');
 var TChannelEndpointHandler = require('../endpoint-handler');
 
+var MAX_RELAY_AD_ATTEMPTS = 2;
```
Comment (jcorbin) [inline node/hyperbahn/handler.js]: even if we define defaults here, can we then use them as an option / self property later on?

File: node/hyperbahn/handler.js [hunk 2]
```diff
@@ -193,27 +196,41 @@ HyperbahnHandler.prototype.sendRelayAdvertise =
 function sendRelayAdvertise(hostPort, services, callback) {
     var self = this;
 
-    self.tchannelJSON.send(self.channel.request({
-        host: hostPort,
-        serviceName: 'hyperbahn'
-    }), 'relay-ad', null, {
-        services: services
-    }, onRelayAdvertise);
+    var attempts = 0;
```
Comment (jcorbin) [inline node/hyperbahn/handler.js]: prefer this to be passed as function arg... could even then hoist it to a method, so we have:
```
HH#sendRelayAd(h, s, cb) {
  self._tryRelayAd(0, h, s, cb);
};

HH#_tryRelayAd(attempt, h, s, cb) {
  ...
  if (maybe) self._tryRelayAd(attemp + 1, h, s, cb);
  else cb(err, res);
};
```

**Comment by jcorbin** [comment]
:ok:

File: node/hyperbahn/handler.js (hunk 1 above)
Comment (Raynos) [inline node/hyperbahn/handler.js]: :+1:

**Comment by Raynos** [comment]
This code works; I'll clean it up later.

**Pull request merged**